### PR TITLE
Deprecate configurations for auth provider configuration on Airflow

### PR DESCRIPTION
**Pull request by nahuelverdugo**
### Describe your changes :
Deprecate the use of the configuration of the bot auth mechanism in the YAML configuration of the OM server. It still allows using it but does not force the user to remove it. 

### Type of change :
- [x] Improvement

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: conf/openmetadata.yaml [hunk 1]
```diff
@@ -181,32 +181,6 @@ airflowConfiguration:
   username: ${AIRFLOW_USERNAME:-admin}
   password: ${AIRFLOW_PASSWORD:-admin}
   metadataApiEndpoint: ${SERVER_HOST_API_URL:-http://localhost:8585/api}
-  authProvider: ${AIRFLOW_AUTH_PROVIDER:-"no-auth"} # Possible values are "no-auth", "azure", "google", "okta", "auth0", "custom-oidc", "openmetadata"
```
Comment (harshach) [inline conf/openmetadata.yaml:184]: we don't need authProvider for service accounts any more?

Comment (nahuelverdugo) [reply conf/openmetadata.yaml:184]: No, we get it now from the Bot auth mechanism.

**Comment by TeddyCr** [comment]
@nahuelverdugo, just wondering what is the status of this PR?

**Comment by nahuelverdugo** [comment]
@harshach can we merge these changes? I think we can stop supporting these configs in 1.0.

**Comment by harshach** [comment]
@nahuelverdugo as long as we don't need any migrations lets merge in. My understanding is if the configs are there we won't throw an error. What happens to existing SSO based service accounts will they work?

**Comment by nahuelverdugo** [comment]
> @nahuelverdugo in 1.0 we can even kill any client auth that's not JWT

@pmbrull should I create a ticket for that?

> @nahuelverdugo as long as we don't need any migrations lets merge in. My understanding is if the configs are there we won't throw an error. What happens to existing SSO based service accounts will they work?

@harshach, we are still supporting them so we won't throw an error. At the start of the application, we check if bots have auth config. If they don't have it, the service accounts configured in the YAML configuration file will be used as the default auth config for bots and stored in the DB. Otherwise, we will use the JWT token config.

**Comment by nahuelverdugo** [comment]
@akash-jain-10 should we remove the Global Chart Values from Helm `global.airflow.openmetadata.authConfig.*` and `global.airflow.openmetadata.authProvider` too?

**Review by harshach — APPROVED** [review]

**Pull request merged**